src/graphics/sprite.js

```diff
@@ -205,10 +201,20 @@ Crafty.c("Sprite", {
     init: function () {
         this.__trim = [0, 0, 0, 0];
         this.bind("Draw", this._drawSprite);
+        this.bind("LayerAttached", this._setupSpriteImage);
     },
 
     remove: function(){
         this.unbind("Draw", this._drawSprite);
+        this.unbind("LayerAttached", this._setupSpriteImage);
+    },
+    
+    _setupSpriteImage: function(layer) {
+        if (!this.__image || !this.img || !layer) return;
```

Comment:
`_setupSpriteImage` and `_setupImage` do the same type of check to make sure the method is only run if both
- the entity is attached to a layer
- the resource to render has loaded